detoks/src/main/java/nl/tudelft/trustchain/detoks/Recommender.kt

```diff
@@ -0,0 +1,70 @@
+package nl.tudelft.trustchain.detoks
+
+import android.util.Log
+import kotlin.random.Random
+
+/**
+ * Basic structure for a profile entry
+ * NOTE: Both the duration and freshness are currently not used
+ */
+class ProfileEntry(
+    var duration: Long = 0,
+    var watchTime: Long = 0,
+    val firstSeen: Long = 0
+)
+
+class Profile(
+    val magnets: HashMap<String, ProfileEntry> = HashMap()
+) {
+    fun updateEntryWatchTime(torrent: TorrentManager.TorrentHandler, time: Long) {
+        val name = torrent.torrentName + "[" + torrent.fileName + "]"
+        if(!magnets.contains(name)) magnets[name] =
+            ProfileEntry(0, time, System.currentTimeMillis())
+        magnets[name]!!.watchTime += time
+        Log.i("DeToks", "Updated watchtime of $name to ${magnets[name]!!.watchTime}")
+    }
+
+    /**
+     * Update the entry duration in the profile
+     * NOTE: This function is currently not used
+     */
+    fun updateEntryDuration(torrent: TorrentManager.TorrentHandler, duration: Long) {
+        val name = torrent.torrentName + "[" + torrent.fileName + "]"
+        if(!magnets.contains(name)) magnets[name] =
+            ProfileEntry(duration, 0, System.currentTimeMillis())
+        else magnets[name]!!.duration = duration
+        Log.i("DeToks", "Updated duration of $name to ${magnets[name]!!.duration}")
+    }
+}
+
+class Recommender {
+    private fun coinTossRecommender(URIs: List<String>): String {
+        return URIs[Random.nextInt(URIs.size)]
+    }
+
+    // TODO: Return a sorted list of videos not just the 'winner'
+    private fun watchTimeDurationRatioRecommender(
+        URIs: List<String>,
+        profiles: List<Profile>
+    ): String {
+        var (candidate, averagePercentageWatched) = Pair("", 0.0f)
+        for (URI: String in URIs) {
+            var (hits, percentageWatchedTotal) = Pair(0.0f, 0.0f)
+            for (profile: Profile in profiles) {
+                if (profile.magnets.containsKey(URI)) {
+                    val entry = profile.magnets[URI]
+                    percentageWatchedTotal += 100.0f * (entry!!.watchTime / entry.duration)
```

Comment:
Is the duration now used or is this still placeholder? I thought the duration was hard to read from a torrent, seeing that the function `updateTorrentDuration` is unused.

Reply:
Removed duration as we do not use it